### PR TITLE
Sync Template Tags with Parameters in Action Creator

### LINKED ISSUE #26437: [P0] Editing action parameters in action editor does not update the parameters under-the-hood

### PR DESCRIPTION
**Pull request by iethree**
closes #26437 

## Description

When editing an already saved action, adding new parameters would apparently update the form (because it was looking at template tags), but actual forms (which use parameters) would not update.

This changes all the forms code to use parameters instead of template tags, and correctly propagates template tags to parameters.

Also updates parameters code to use the `display-name` in forms.

## Demo

When? | What?
--- | ---
Before | paramRepro
After | fixDemo

### CODE REVIEW

File: frontend/src/metabase/writeback/components/ActionCreator/QueryActionEditor.tsx [hunk 1]
```diff
@@ -16,9 +16,14 @@ export function QueryActionEditor({
...
         viewHeight="full"
-        setDatasetQuery={(newQuery: Query) =>
-          setQuestion(question.setQuery(newQuery))
-        }
+        setDatasetQuery={(newQuery: NativeQuery) => {
+          // we need to sync the query AND the template tags
+          setQuestion(
+            question
+              .setQuery(newQuery)
+              .setParameters(newQuery?.templateTagsWithoutSnippets()),
```
Comment (iethree) [inline frontend/src/metabase/writeback/components/ActionCreator/QueryActionEditor.tsx:24]: This is the crucial bit. Honestly, I expected that updating the query would update the parameters which are derived from the template tags in the query 🤷

File: frontend/src/metabase/writeback/components/ActionCreator/FormCreator/FormCreator.tsx [hunk 2]
```diff
@@ -40,48 +40,50 @@ export function FormCreator({
...
   onChange,
 }: {
-  tag: TemplateTag;
+  param: Parameter;
   fieldSettings: FieldSettings;
   onChange: (fieldSettings: FieldSettings) => void;
 }) {
   const [isEditingOptions, setIsEditingOptions] = useState(false);
-  const name = tag.name;
+  const name = param["display-name"] ?? param.name;
```
Comment (iethree) [inline frontend/src/metabase/writeback/components/ActionCreator/FormCreator/FormCreator.tsx:86]: this is unrelated, but while I was here, I made field names display prettier.

**Review by kulyk — COMMENTED** [review]
@iethree it seems that it made impossible to map action parameters to page filters, please take a look at the repro:

https://user-images.githubusercontent.com/17258145/202200616-a519cfd5-790c-4fa0-ae2e-fe381b23f1d8.mp4

File: frontend/src/metabase/writeback/components/ActionCreator/QueryActionEditor.tsx (hunk 1 above)
Comment (iethree) [reply frontend/src/metabase/writeback/components/ActionCreator/QueryActionEditor.tsx:24]: paranoia 😨

**Comment by iethree** [comment]
> @iethree it seems that it made impossible to map action parameters to page filters, please take a look at the repro:
> 
>  CleanShot.2022-11-16.at.14.00.14.mp4

Thanks for catching that 😳 . Took me a lot longer than I care to admit to come up with the fix: e526b8255b5d6c768b89dcec710ea4331e748470

**Review by kulyk — APPROVED** [review]
Looks good! Another thing I noticed, but doesn't look to blocking: seems that I can map a page filter only to a single action parameter. The moment I want to map another parameter to the same filter, the previous parameter mapping resets and becomes "Ask a user"

https://user-images.githubusercontent.com/17258145/202449614-d1aa8a8c-0234-4fe1-aa0e-671592fc9b36.mp4

**Comment by iethree** [comment]
> Looks good! Another thing I noticed, but doesn't look to blocking: seems that I can map a page filter only to a single action parameter. The moment I want to map another parameter to the same filter, the previous parameter mapping resets and becomes "Ask a user"
> 
>  CleanShot.2022-11-17.at.12.43.07.mp4

This is an existing limitation of `variable` type parameters that we inherited from text cards.  maybe we should rethink if this makes sense for our case 🤔

**Pull request merged**